pennylane/templates/swap_networks/ccl2.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the TwoLocalSwapNetwork template.
+"""
+
+import warnings
+import pennylane as qml
+import pennylane.numpy as np
+from pennylane.operation import Operation, AnyWires
+from pennylane.ops import SWAP, FermionicSWAP
+
+
+class TwoLocalSwapNetwork(Operation):
+    r"""Apply two-local gate operations using a canonical 2-complete linear (2-CCL) swap network.
+
+    Args:
+        wires (Iterable or Wires): ordered sequence of wires on which the swap network acts
+        acquaintances (Callable): callable `func(index, wires, param=None, **kwargs)` that returns a two-local operation applied on a pair of logical wires specified by `index` currently stored in physical wires provided by `wires` before they are swapped apart. Parameters for the operation are specified using `param`, and any additional keyword arguments for the callable should be provided using the ``kwargs`` separately
```

Comment:
I find this description hard to understand. Instead of calling them logical and physical wires, can we be more explicit about what they refer to? Consider renaming `index` and `wires` to some short version of  `current_wire_indices` and `original_wire_indices`. That way their meaning is clearer. Everything has been swapped around, so `current_wire_indices` are the wires you have to act on if you want to operate on `original_wire_indices`. There should be some way to express this clearly. I tried unsuccessfully below, but maybe the suggestions helps.
```suggestion
        acquaintances (Callable): callable `func(index, wires, param=None, **kwargs)` that returns a two-wire operation. This operation is applied on a pair of wires specified by `index`. But this corresponds to applying the operation on `wires` before any SWAP gates occurred. Parameters for the operation are specified using `param`, and any additional keyword arguments for the callable should be provided using the ``kwargs`` separately
```